src/arns.lua

```diff
@@ -99,7 +107,16 @@ function arns.extendLease(from, name, years, currentTimestamp)
 	-- Transfer tokens to the protocol balance
 	balances.transfer(ao.id, from, totalExtensionFee)
 	demand.tallyNamePurchase(totalExtensionFee)
-	return arns.getRecord(name)
+	return {
+		record = arns.getRecord(name),
+		totalExtensionFee = totalExtensionFee,
+		baseRegistrationFee = baseRegistrationFee,
+		remainingBalance = balances.getBalance(from),
+		protocolBalance = balances.getBalance(ao.id),
+		recordsCount = utils.lengthOfTable(NameRegistry.records),
+		reservedRecordsCount = utils.lengthOfTable(NameRegistry.reserved),
+		df = demand.getDemandFactorInfo(),
```

Comment:
do these need to be returned by this function? you can get them directly in main.lua

Reply:
Sure I can make that change :)